src/resources/RepositoryResource.ts

```diff
@@ -54,6 +54,31 @@ export async function fetchRepositoriesForNamespace(ns: string) {
   return response.data?.repositories;
 }
 
+export interface RepositoryDetails {
+  can_admin: boolean;
+  can_write: boolean;
+  description: string | null;
+  is_free_account: boolean;
+  is_organization: boolean;
+  is_public: boolean;
+  is_starred: boolean;
+  kind: string | null;
+  name: string | null;
+  namespace: string | null;
+  state: string | null;
+  status_token: string | null;
+  tag_expiration_s: number | null;
+  trust_enabled: boolean;
+}
+
+export async function fetchRepositoryDetails(org: string, repo: string) {
+  const response: AxiosResponse<RepositoryDetails> = await axios.get(
+    `/api/v1/repository/${org}/${repo}?includeStats=false&includeTags=false`,
+  );
+  assertHttpCode(response.status, 200);
```

Comment:
not sure if you need this as axios should throw an error on any non 2xx code

Reply:
Yeah this was just to confirm the exact expected code. We probably don't need to do this for the reason you mentioned. We can talk about removing  itduring the refactor post MVP.